modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -344,6 +344,58 @@ public void asyncDestroyService(Deployment deployment, DeployTask deployTask) {
 
     }
 
+    /**
+     * purge the service based on the serviceDeploymentState.
+     *
+     * @param deployment deployment
+     * @param deployTask deployTask
+     */
+    public void purgeService(Deployment deployment, DeployTask deployTask) {
```

Comment:
can you also please add tests for this? one for success case, one for rejection case.